### PR TITLE
feat(dive-log): linear estimated tank pressure line for manual dives (#197)

### LINKED ISSUE #197: When no AI pressure data exists for a defined tank, show linear graph on dive profile chart
When no AI pressure data exists for a defined tank, graph the linear start to end pressures from defined tanks in the dive profile chart.

### PR DESCRIPTION
**Pull request by ericgriffin**
## Summary

Closes #197. When a dive has a tank with manually-entered start/end pressures but **no air-integrated (AI) transmitter data**, the profile chart previously drew no pressure line at all. This adds a linear **estimated** pressure line so manual-entry dives still get a pressure visualization.

The line is shaped like a real transmitter would have recorded it: flat at the start pressure while the tank is unused, dropping linearly while it is breathed (windowed by gas switches), flat during gaps, and flat at the end pressure afterward. The total drop is distributed across the tank's active windows in proportion to their duration. For the common single-tank dive this collapses to one straight `start → end` line.

Synthesis happens **in memory at the chart-read path and is never persisted** — SAC analysis and UDDF/other exports read the repository directly, so they continue to see real measured data only. The estimated line is drawn straight (real AI lines stay curved) and marked "(est.)" in the legend, tooltip, and readout so an interpolation is never mistaken for a transmitter trace.

Design spec and implementation plan: `docs/superpowers/specs/2026-07-06-linear-tank-pressure-line-design.md`, `docs/superpowers/plans/2026-07-06-linear-tank-pressure-line.md`.

## Changes

- **`buildActiveTankIntervals`** (`gas_usage_segments_service.dart`) — per-tank active `[start,end)` windows from gas switches, reusing the same starting-tank + switch-walk rules as `buildGasUsageSegments` (keyed by tankId, no gas-mix merging).
- **`synthesizeEstimatedTankPressures` + `EstimatedTankPressures`** (new `estimated_tank_pressure_synthesizer.dart`) — pure flat–drop–flat series builder with duration-proportional drop distribution and full-dive fallbacks; only synthesizes when a tank has both pressures, `start > end`, no real time-series data, and the dive has a profile.
- **`estimatedTankPressuresProvider`** (`dive_providers.dart`) — composes real pressures + estimates; chart-only.
- **Chart** (`dive_profile_chart.dart`) — new `estimatedTankIds` field; estimated lines rendered `isCurved: false`; "(est.)" suffix in tooltip and draggable-readout rows.
- **Legend** (`dive_profile_legend.dart`) — "(est.)" suffix on estimated tank rows in the Tank Pressures section.
- **Hosts** (`dive_profile_panel`, `dive_detail_page`, `fullscreen_profile_page`) — pass the augmented map + `estimatedTankIds` to the chart only; SAC-normalization and marker builders keep reading the real map.
- **i18n** — new `diveLog_pressure_estimatedSuffix` string translated across all 11 locales.

## Test Plan

- [x] `flutter analyze` passes (whole project, locally clean)
- [x] `flutter test test/features/dive_log/` passes (1825 tests; full suite runs in CI)
- New coverage: interval derivation, synthesizer trigger/shape/edge cases (incl. the two spec worked examples), provider composition, straight-line rendering, and the "(est.)" label in tooltip/readout/legend.
- [ ] Manual testing on: **macOS deferred** (branch built in a checkout shared with a concurrent session). Recommend a visual check on a real manual-entry dive before merge.

### CODE REVIEW

File: lib/features/dive_log/data/services/estimated_tank_pressure_synthesizer.dart [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+    if (i + 1 < sorted.length && sorted[i + 1].start > w.end) {
+      points.add(_pt(tankId, sorted[i + 1].start, pressure));
+    }
+  }
+  if (sorted.last.end < diveDurationSeconds) {
+    points.add(_pt(tankId, diveDurationSeconds, endPressure));
+  } else {
+    // Clamp the final vertex to endPressure to absorb floating-point drift.
+    points[points.length - 1] = _pt(tankId, sorted.last.end, endPressure);
+  }
```
[the remark this answers is not in the document]
Comment (ericgriffin) [reply lib/features/dive_log/data/services/estimated_tank_pressure_synthesizer.dart:98]: Fixed in 5562c7a. The last window-end vertex is now clamped to `endPressure` inside the loop (previously only the `last.end == diveDurationSeconds` branch was clamped), so the flat tail stays horizontal regardless of accumulated drift. Added a regression test with empirically-verified drifting parameters — 210→100 over a 300s window yields 100.00000000000001 pre-fix.

File: lib/features/dive_log/presentation/providers/dive_providers.dart [hunk 2]
```diff
@@ -958,6 +960,28 @@ final tankPressuresProvider =
...
+      }
+      return synthesizeEstimatedTankPressures(
+        existing: real,
+        tanks: dive.tanks,
+        gasSwitches: switches,
+        diveDurationSeconds: dive.profile.isEmpty
+            ? 0
+            : dive.profile.last.timestamp,
+      );
+    });
```
[the remark this answers is not in the document]
Comment (ericgriffin) [reply lib/features/dive_log/presentation/providers/dive_providers.dart:988]: Fixed in 20ccfb0. The three fetches now start concurrently (watch all futures, then await) instead of waterfalling, and `switches` is awaited only after confirming `dive != null`, so the null path does no extra work.
